lib/traject/config/sirsi_config.rb

```diff
@@ -3023,6 +3023,24 @@ def library_has(field)
   end
 end
 
+##
+# IIIF Manifest field based on the presense of "file", "*.jp2" match in an
+# 856 SDR-PURL entry
+to_field 'iiif_manifest_url_ssim' do |record, accumulator|
+  Traject::MarcExtractor.new('856x').collect_matching_lines(record) do |field, spec, extractor|
+    subfields = extractor.collect_subfields(field, spec)
+    next unless subfields[0] == 'SDR-PURL' && subfields[1] == 'item'
+
+    accumulator.concat(subfields.slice(2..-1).map do |v|
+      v.split(':')
+    end.select do |(type, file_id)|
+      type == 'file' && /.*\.jp2/.match(file_id)
+    end.map do |(_type, file_id)|
```

Comment:
I think this is fine.  I'm not exactly sure what we would do w/ multiple IIIF manifests in the same doc, but not really a concern of this code.